6_drivers_prep_rmk.yml

```diff
@@ -0,0 +1,59 @@
+target_default: 6_drivers_prep
+
+packages:
+  - sp
+  - scipiper
+  - dplyr
+
+sources:
+  - 6_drivers_prep/code/nldas_cube_utils.R
+  - 6_drivers_prep/code/nldas_feather_utils.R
+  
+targets:
+  6_drivers_prep: 
+    depends: 
+      - 7_nldas_subset_rmk.yml
+      - 7_nldas_append_rmk.yml
+  
+  nldas_subset_time:
+    command: c(t0=I(0), t1 = I(145551))#345551
+    
+  nldas_time_chunk:
+    command: c(I(5000))
+    
+  nldas_cells:
+    command: list(x=I(c(221,221,344,344,223)), y=I(c(132,196,132,196,141)), variables=I(c('dlwrfsfc','pressfc','apcpsfc','vgrd10m','ugrd10m','dswrfsfc','spfh2m','tmp2m'))) # will come from code or spatial data...
+      
+  nldas_subset_box: # this is a FULL time pull
+    command: nldas_diff_box(nldas_cells, '6_drivers_prep/in/nldas_cells.rds') 
+      
+  nldas_append_box: # this is a PARTIAL time pull, where partial time could be 0 (complete)
```

Comment:
How do the full and partial time pulls work together? In the first round of data pulling? In subsequent rounds? I guess I don't really (yet) understand the meaning of "subset" vs "append" throughout this file.

Reply:
Yes, I wrote this to make space to be able to either add new sites that expand the bounding box of the cells that are pulled or new time steps that would be added to existing files. 

"subset" is a full data pull for all time that is specified.
"append" is a data pull for only partial time (it won't pull time steps you already have). 

"append" is really rare initially, because all of the pulls are happening for the first time, but if we wanted to re-pull things next year with a longer time dimension, I think all of the tasks would end up being "append" jobs (unless we also updated the spatial domain at the same time).

Reply:
ahh, OK, so "subset" targets are also responsible for appending to the data, but in a spatial rather than temporal dimension?

Reply:
yes. I think "append" is an OK word for what those targets do, but "subset" isn't a great choice. I couldn't think of a better word and moved on and kept it for the time being

Reply:
i know it adds underscores, but maybe `append_time` and `append_sites` (or `add_time` and `add_sites`)?

Reply:
I think I am leaning towards the following:
```yaml
  nldas_replete_box: # this is the domain that will require a FULL data pull in the time dimension
    command: nldas_diff_box(nldas_cells, '6_drivers_prep/in/nldas_cells.rds') 

  nldas_partial_box: # this is the domain that will require a PARTIAL time pull, where partial time could be 0 (complete)
    command: nldas_update_box(nldas_cells, nldas_replete_box) 
    
  nldas_partial_time:
    command: nldas_times_diff(nldas_time_range, '6_drivers_prep/in/nldas_times.rds')

  nldas_new_cube_files:
    command: calc_nldas_files(nldas_replete_box, nldas_time_range, nldas_time_chunk)
  
  nldas_append_cube_files:
    command: calc_nldas_files(nldas_partial_box, nldas_partial_time, nldas_time_chunk)
  
  nldas_new_cells:
    command: nldas_diff_cells(nldas_cells, '6_drivers_prep/in/nldas_cells.rds') 
    
  nldas_append_cells:
    command: nldas_update_cells(nldas_cells, nldas_new_cells, '6_drivers_prep/in/nldas_times.rds')
```

Reply:
Cool, I like the new vs append distinction. I like the sound of replete and partial...but the replete box describes the cells where currently there is NO data, so from one perspective it seems backward, and I could almost go for `nldas_empty_box` instead...that said, in combination with the comments, I'm having no trouble understanding what you're doing now. And I can't think of a better naming scheme for the nldas_xx_box targets, so I say go for it.